From 533811deef3155abe71dbace6960feee0aa8a35a Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Wed, 31 Jul 2019 19:09:17 +0200
Subject: [PATCH] user: Support devices with more than 5 enroll steps

We are currently adding support for Synaptics devices that require 8
steps. Add another row for images which brings us to up to 10
supportable steps for now.
---
 .../user-accounts/data/account-fingerprint.ui | 62 +++++++++++++++++++
 panels/user-accounts/um-fingerprint-dialog.c  |  2 +-
 2 files changed, 63 insertions(+), 1 deletion(-)

diff --git a/panels/user-accounts/data/account-fingerprint.ui b/panels/user-accounts/data/account-fingerprint.ui
index e352e6de1..969f7ca30 100644
--- a/panels/user-accounts/data/account-fingerprint.ui
+++ b/panels/user-accounts/data/account-fingerprint.ui
@@ -243,6 +243,68 @@
             <property name="position">1</property>
           </packing>
         </child>
+        <child>
+          <object class="GtkHBox" id="enroll2_hbox">
+            <property name="visible">True</property>
+            <property name="margin-top">24</property>
+            <property name="margin-bottom">24</property>
+            <child>
+              <object class="GtkImage" id="image6">
+                <property name="visible">True</property>
+                <property name="stock">gtk-no</property>
+                <property name="icon-size">6</property>
+              </object>
+              <packing>
+                <property name="position">1</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkImage" id="image7">
+                <property name="visible">True</property>
+                <property name="stock">gtk-no</property>
+                <property name="icon-size">6</property>
+              </object>
+              <packing>
+                <property name="position">2</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkImage" id="image8">
+                <property name="visible">True</property>
+                <property name="stock">gtk-no</property>
+                <property name="icon-size">6</property>
+              </object>
+              <packing>
+                <property name="position">3</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkImage" id="image9">
+                <property name="visible">True</property>
+                <property name="stock">gtk-no</property>
+                <property name="icon-size">6</property>
+              </object>
+              <packing>
+                <property name="position">4</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkImage" id="image10">
+                <property name="visible">True</property>
+                <property name="stock">gtk-no</property>
+                <property name="icon-size">6</property>
+              </object>
+              <packing>
+                <property name="position">5</property>
+              </packing>
+            </child>
+          </object>
+          <packing>
+            <property name="expand">False</property>
+            <property name="fill">False</property>
+            <property name="position">2</property>
+          </packing>
+        </child>
         <child>
           <object class="GtkLabel" id="status-label">
             <property name="visible">True</property>
diff --git a/panels/user-accounts/um-fingerprint-dialog.c b/panels/user-accounts/um-fingerprint-dialog.c
index 48b12dcf3..f5dee5b95 100644
--- a/panels/user-accounts/um-fingerprint-dialog.c
+++ b/panels/user-accounts/um-fingerprint-dialog.c
@@ -32,7 +32,7 @@
 #include "fingerprint-strings.h"
 
 /* This must match the number of images on the 2nd page in the UI file */
-#define MAX_ENROLL_STAGES 5
+#define MAX_ENROLL_STAGES 10
 
 static GDBusProxy *manager = NULL;
 static GDBusConnection *connection = NULL;
-- 
2.24.1

